src/main/java/org/folio/rest/impl/PatronBlockConditionsAPI.java

```diff
@@ -0,0 +1,135 @@
+    Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+
+    PgUtil.get(PATRON_BLOCK_CONDITIONS, PatronBlockCondition.class,
+      org.folio.rest.jaxrs.model.PatronBlockConditions.class, query, offset, limit,
+      okapiHeaders, vertxContext, GetPatronBlockConditionsResponse.class, asyncResultHandler);
+  }
+
+  @Validate
+  @Override
+  public void postPatronBlockConditions(String lang, PatronBlockCondition entity,
+    Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
+
+    Errors errors = isEntityValid(entity, asyncResultHandler);
+    if (errors != null) {
+      asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(
+        PatronBlockConditions.PostPatronBlockConditionsResponse
+          .respond422WithApplicationJson(errors)));
+    }
+
+    PgUtil.post(PATRON_BLOCK_CONDITIONS, entity, okapiHeaders, vertxContext,
+      PostPatronBlockConditionsResponse.class, asyncResultHandler);
+  }
+
+  @Validate
+  @Override
+  public void putPatronBlockConditionsByPatronBlockConditionId(
+    String patronBlockConditionId, String lang, PatronBlockCondition entity,
+    Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
+
+    Errors errors = isEntityValid(entity, asyncResultHandler);
+    if (errors != null) {
+      asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(
+        PatronBlockConditions.PutPatronBlockConditionsByPatronBlockConditionIdResponse
+          .respond422WithApplicationJson(errors)));
+    }
+
+    PgUtil.put(PATRON_BLOCK_CONDITIONS, entity, patronBlockConditionId, okapiHeaders,
+      vertxContext, PutPatronBlockConditionsByPatronBlockConditionIdResponse.class, asyncResultHandler);
+  }
+
+  @Validate
+  @Override
+  public void getPatronBlockConditionsByPatronBlockConditionId(
+    String patronBlockConditionId, String lang, Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+
+    PgUtil.getById(PATRON_BLOCK_CONDITIONS, PatronBlockCondition.class, patronBlockConditionId,
+      okapiHeaders, vertxContext, GetPatronBlockConditionsByPatronBlockConditionIdResponse.class,
+      asyncResultHandler);
+  }
+
+  @Validate
+  @Override
+  public void deletePatronBlockConditionsByPatronBlockConditionId(
+    String patronBlockConditionId, String lang, Map<String,
+    String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
+
+    PgUtil.deleteById(PATRON_BLOCK_CONDITIONS, patronBlockConditionId, okapiHeaders,
+      vertxContext, DeletePatronBlockConditionsByPatronBlockConditionIdResponse.class,
+      asyncResultHandler);
+  }
+
+  private Errors isEntityValid(PatronBlockCondition entity, Handler<AsyncResult<Response>> asyncResultHandler) {
+    Errors errors = null;
+    if (!isEntityMessageValid(entity)) {
```

Comment:
Can you avoid the negations and the additional method by directly writing what the error message says?

    if (isMessageBlank(entity) && isAnyFlagTrue(entity)) {

Reply:
I'm afraid not because I should define 2 different messages for two cases: 

If Borrowing and/or Renewals and/or Requests flag is set to true, then Message to be displayed is required. Return an error with this message if it is not present: **Message to be displayed is a required field if one or more Blocked actions selected**
If Message to be displayed is present, then Borrowing and/or Renewals and/or Requests must be set to true. Return an error with this message if no action flag is set to true: O**ne or more Blocked actions must be selected for Message to be displayed to be used**

Here I should exactly know what was missed: message or a flag. The condition `(isMessageBlank(entity) && isAnyFlagTrue(entity))` will not allow to define appropriate error message.

Reply:
Works for me: https://github.com/folio-org/mod-users/pull/162/files
Can you add a unit test that shows that this is wrong?

Reply:
Thanks, you are right. 
I've updated tests to check the error message, the behavior was not changed.
Updated